flycheck.el

```diff
@@ -3685,7 +3690,16 @@ Intended for use with `next-error-function'."
   (let ((pos (flycheck-next-error-pos n reset)))
     (if pos
         (goto-char pos)
-      (user-error "No more Flycheck errors"))))
+      (let ((result (cl-remove-if
+                     (lambda (err)
+                       (eq (current-buffer)
+                           (flycheck-error-buffer err)))
+                     flycheck-current-errors)))
+        (if result
+            (let ((error (car result)))
+              (flycheck-jump-to-error error)
+              (flycheck-buffer))
+          (user-error "No more Flycheck errors"))))))
```

Comment:
I think this goes to the first error not in the current buffer (the idea being that once you reach the bottom of errors in the current buffer, you'll want to move to the next one).  I agree that this could use a bit of extra clarification :)